### PR TITLE
fix: update rather than replace existing cache entry

### PR DESCRIPTION
**Pull request by edreed**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Since `TimedCache` relies on a per-entry `sync.Mutex` to serialize calls to the refresh function, `TimedCache.Update` **must** update the existing entry rather than replace it to ensure proper synchronization. Otherwise, it is possible for multiple concurrent calls to the refresh function to occurr.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fixes an issue in the `TimedCache` implementation that could result in multiple concurrent calls to the getter function.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/cache/azure_cache.go [hunk 1]
```diff
@@ -189,9 +189,16 @@ func (t *TimedCache) Set(key string, data interface{}) {
 
 // Update updates the data cache for the key.
 func (t *TimedCache) Update(key string, data interface{}) {
-	_ = t.Store.Update(&AzureCacheEntry{
```
Comment (feiskyer) [inline pkg/cache/azure_cache.go:192]: Could you add some unit tests to catch this condition?

Comment (edreed) [reply pkg/cache/azure_cache.go:192]: Done.

**Comment by andyzhangx** [comment]
@edreed could you fix the lint issue? after this PR merged, I will publish a new release late this week, thanks.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**